debezium-core/src/main/java/io/debezium/transforms/TimezoneConverter.java

```diff
@@ -0,0 +1,388 @@
+
+    private Struct handleEnvelopeValue(Schema schema, Struct value) {
+        final Struct updatedEnvelopeValue = new Struct(schema);
+        Struct updatedAfterValue = getStruct(value, Envelope.FieldName.AFTER);
+        Struct updatedBeforeValue = getStruct(value, Envelope.FieldName.BEFORE);
+        if (updatedAfterValue != null) {
+            updatedEnvelopeValue.put(Envelope.FieldName.AFTER, updatedAfterValue);
+        }
+        if (updatedBeforeValue != null) {
+            updatedEnvelopeValue.put(Envelope.FieldName.BEFORE, updatedBeforeValue);
+        }
+        return updatedEnvelopeValue;
+    }
+
+    private Struct getStruct(Struct struct, String structName) {
+        try {
+            return Requirements.requireStructOrNull(struct.getStruct(structName), "");
+        }
+        catch (DataException ignored) {
+        }
+        return null;
+    }
+
+    private String getTableFromSource(Struct value) {
+        try {
+            Struct source = value.getStruct(Envelope.FieldName.SOURCE);
+            return source.getString("table");
+        }
+        catch (DataException ignored) {
+        }
+        return null;
+    }
+
+    private static class FieldItem {
+        private final String prefix;
+        private final String matchName;
+        private final String fieldName;
+
+        FieldItem(String prefix, String matchName, String fieldName) {
+            this.prefix = prefix;
+            this.matchName = matchName;
+            this.fieldName = fieldName;
+        }
+
+        public String getPrefix() {
+            return prefix;
+        }
+
+        public String getMatchName() {
+            return matchName;
+        }
+
+        public String getFieldName() {
+            return fieldName;
+        }
+    }
+
+    private FieldItem parseItem(String item) {
+        String prefix = null;
+        String matchName = null;
+        String fieldName = null;
+
+        String[] parts = item.split(":");
+        if (parts.length == 1) {
+            // table or topic
+            matchName = parts[0];
+        }
+        else if (parts.length >= 2 && parts.length <= 3) {
+            if (parts[0].equalsIgnoreCase(SOURCE) || parts[0].equalsIgnoreCase(TOPIC)) {
+                // source:table or topic:topic
+                // source:table1,table2 or topic:topic1,topic2
+                prefix = parts[0];
+                matchName = parts[1];
+                if (parts.length == 3) {
+                    // source:table:field or topic:topic:field
+                    fieldName = parts[2];
+                }
+            }
+            else {
+                // table:field or topic:field
+                matchName = parts[0];
+                fieldName = parts[1];
+            }
+        }
+        return new FieldItem(prefix, matchName, fieldName);
+    }
+
+    private void handleInclude(Struct value) {
+        for (String item : includeList) {
+            FieldItem includeItem = parseItem(item);
```

Comment:
Is it possible to do all parsing in advance in config method and prepare filtering logic/structures in advance?